Chatterjee's Xi returns different results based on `numpy` major version

### Describe your issue.

# Issue
Depending on the input data, the coefficient and p-value results of running `scipy.stats.chatterjeexi()` often returns different results based on whether the version of `numpy` installed is >=2.0.0 or <2.0.  Those using `numpy>=2.0` seem to be more correct, but this has not been verified. This is unusual because other correlation functions' coefficients, like Pearson's R and Spearman's Rho, don't have changing coefficients based on numpy's version. And, according to [Chatterjee (2021)](https://doi.org/10.1080/01621459.2020.1758115), this coefficient should not either. 

The _Xi_ coefficient should be between 0-1, but using a proprietary dataset (that I cannot share here) of 14,366 observations, the results using the identical data from the same dataset in the same order are as follows:

> Scipy v1.16.0 with Numpy at v1.26.4 ran Chatterjee's Xi and got a corr=1.2256     w/ P-value=0.0

versus

> Scipy with Numpy at v2.0.0 ran Chatterjee's Xi and got a corr=0.2087*** w/ P-value=0.0

The only variable changed here is the numpy version. The latter with v2.0.0 seems more appropriate, both because it is within the range Chatterjee sets for appropriate calculation of his coefficient, and it lines up a bit better with Spearman's Rho coefficient for the same two variables. 


# Cause
My best guess, based on my limited testing, is that this arises from the following code in scipy.stats._correlation.py:

https://github.com/scipy/scipy/blob/9ae478ddaec6e1a71ca31624199e1ea171835eec/scipy/stats/_correlation.py#L14-L24

The `numpy==2.0.0` [release notes](https://github.com/numpy/numpy/releases/tag/v2.0.0#:~:text=Minor%20changes%20in%20behavior%20of%20sorting%20functions) detail that 

> methods that aren't stable may return slightly different results in 2.0.0 compared to 1.26.x. This includes the default method of numpy.argsort. 

Testing shows that the Xi statistic's computing using numpy's argsort, broadcast_arrays, and take_along_axis all differ based on numpy's version as 1.26.x vs 2.0.0. Because _Xi_ is a rank-order coefficient, my hunch is that the ordering of one or all of these functions differs because of the introduction of `stable` in numpy v2.0 to the `argsort` function. I'm not sophisticated enough to thoroughly test this, though. 

# Notes

- Chatterjee's Xi was introduced in `scipy v1.15.0`, https://github.com/scipy/scipy/issues/21453. 
- I included a test dataset below that will reproduce the error, albeit with oddly negative coefficients. I'm aware that mathematically, this can happen, as noted in both the paper itself and a competing python implementation of calculating Xi: https://github.com/swarnakumar/xicorpy/issues/32#issuecomment-2253698637. Maybe not the best example dataset(?), but it's still odd and changes the coefficient based on the numpy version.
- Forgive me if issues pertaining to `numpy` versioning are handled differently in this package.

### Reproducing Code Example

```python
import numpy as np
import pandas as pd
from scipy import stats as sps, __version__ as spv

non_iid_df = pd.DataFrame(
    {
        "X": [1, 1, 3, 3, 5, 5, 7, 7, 9, 9, 11, 11, 13, 13, 15, 15, 17, 17, 19, 19, 21, 21, 23, 23, 25, 25, 27, 27, 29, 29, 31, 31, 33, 33, 35, 35, 37, 37, 39, 39, 41, 41, 43, 43, 45, 45, 47, 47, 49, 49, 51, 51, 53, 53, 55, 55, 57, 57, 59, 59, 61, 61, 63, 63, 65, 65, 67, 67, 69, 69],
        "Y": [5, 6, 5, 6, 5, 6, 5, 6, 5, 6, 5, 16, 5, 16, 5, 16, 5, 16, 5, 16, 5, 26, 5, 26, 5, 26, 5, 26, 5, 26, 5, 36, 5, 36, 5, 36, 5, 36, 5, 36, 5, 46, 5, 46, 5, 46, 5, 46, 5, 46, 5, 56, 5, 56, 5, 56, 5, 56, 5, 56, 5, 66, 5, 66, 5, 66, 5, 66, 5, 66]
    }
)

x = non_iid_df["X"]
y = non_iid_df["Y"]

sps_xi_obj = sps.chatterjeexi(
    x=np.asarray(x).astype(np.float64),
    y=np.asarray(y).astype(np.float64),
    nan_policy="omit",
)
sps_xi_cor, sps_xi_p = float(sps_xi_obj.statistic), float(sps_xi_obj.pvalue)
print(
    f"Scipy (version {spv}) with Numpy (version {np.__version__}) ran Chatterjee's Xi and got a corr={round(sps_xi_cor, 5)} w/ P-val={round(sps_xi_p, 5)}\n"
)
```

### Error message

```shell
Scipy (version 1.16.0) with Numpy (version 1.26.4) ran Chatterjee's Xi and got a corr=-0.29444 w/ P-val=0.99953
---------------------
Scipy (version 1.16.0) with Numpy (version 2.0.0) ran Chatterjee's Xi and got a corr=-0.51667 w/ P-val=1.0
```

### SciPy/NumPy/Python version and system information

```shell
sys.version_info(major=3, minor=12, micro=4, releaselevel='final', serial=0)

Build Dependencies:
  blas:
    detection method: pkgconfig
    found: true
    include directory: C:/Users/runneradmin/AppData/Local/Temp/cibw-run-69ol1m2h/cp312-win_amd64/build/venv/Lib/site-packages/scipy_openblas32/include
    lib directory: C:/Users/runneradmin/AppData/Local/Temp/cibw-run-69ol1m2h/cp312-win_amd64/build/venv/Lib/site-packages/scipy_openblas32/lib
    name: scipy-openblas
    openblas configuration: OpenBLAS 0.3.28 DYNAMIC_ARCH NO_AFFINITY Haswell MAX_THREADS=24
    pc file directory: D:/a/scipy/scipy
    version: 0.3.28
  lapack:
    detection method: pkgconfig
    found: true
    include directory: C:/Users/runneradmin/AppData/Local/Temp/cibw-run-69ol1m2h/cp312-win_amd64/build/venv/Lib/site-packages/scipy_openblas32/include
    lib directory: C:/Users/runneradmin/AppData/Local/Temp/cibw-run-69ol1m2h/cp312-win_amd64/build/venv/Lib/site-packages/scipy_openblas32/lib
    name: scipy-openblas
    openblas configuration: OpenBLAS 0.3.28 DYNAMIC_ARCH NO_AFFINITY Haswell MAX_THREADS=24
    pc file directory: D:/a/scipy/scipy
    version: 0.3.28
  pybind11:
    detection method: config-tool
    include directory: unknown
    name: pybind11
    version: 2.13.6
Compilers:
  c:
    commands: cc
    linker: ld.bfd
    name: gcc
    version: 10.3.0
  c++:
    commands: c++
    linker: ld.bfd
    name: gcc
    version: 10.3.0
  cython:
    commands: cython
    linker: cython
    name: cython
    version: 3.1.2
  fortran:
    commands: gfortran
    linker: ld.bfd
    name: gcc
    version: 10.3.0
  pythran:
    include directory: C:\Users\runneradmin\AppData\Local\Temp\build-env-bn1v0hof\Lib\site-packages/pythran
    version: 0.18.0
Machine Information:
  build:
    cpu: x86_64
    endian: little
    family: x86_64
    system: windows
  cross-compiled: false
  host:
    cpu: x86_64
    endian: little
    family: x86_64
    system: windows
Python Information:
  path: C:\Users\runneradmin\AppData\Local\Temp\build-env-bn1v0hof\Scripts\python.exe
  version: '3.12'
```

I'm having trouble building with NumPy 1.26. 

<details>

```
(scipy-dev-numpy126) C:\Users\matth\Desktop\scipy>spin install
$ pip install -v --config-settings=editable-verbose=true --no-build-isolation --editable .
Using pip 25.1.1 from C:\Users\matth\.local\share\mamba\envs\scipy-dev-numpy126\Lib\site-packages\pip (python 3.12)
Obtaining file:///C:/Users/matth/Desktop/scipy
  Running command Checking if build backend supports build_editable
  Checking if build backend supports build_editable ... done
  Running command Preparing editable metadata (pyproject.toml)
  + meson setup C:\Users\matth\Desktop\scipy C:\Users\matth\Desktop\scipy\build\cp312 -Dbuildtype=release -Db_ndebug=if-release -Db_vscrt=md --native-file=C:\Users\matth\Desktop\scipy\build\cp312\meson-python-native-file.ini
  The Meson build system
  Version: 1.8.2
  Source dir: C:\Users\matth\Desktop\scipy
  Build dir: C:\Users\matth\Desktop\scipy\build\cp312
  Build type: native build
  Project name: scipy
  Project version: 1.16.1.dev0+git20250623.f66c2ff
  C compiler for the host machine: cc (gcc 10.3.0 "cc (Built by Jeroen for the R-project) 10.3.0")
  C linker for the host machine: cc ld.bfd 2.36.1
  C++ compiler for the host machine: c++ (gcc 10.3.0 "c++ (Built by Jeroen for the R-project) 10.3.0")
  C++ linker for the host machine: c++ ld.bfd 2.36.1
  Cython compiler for the host machine: cython (cython 3.1.2)
  Host machine cpu family: x86_64
  Host machine cpu: x86_64
  Program python found: YES (C:\Users\matth\.local\share\mamba\envs\scipy-dev-numpy126\python.exe)
  Run-time dependency python found: YES 3.12
  Program cython found: YES (C:\Users\matth\.local\share\mamba\envs\scipy-dev-numpy126\Scripts\cython.EXE)
  Compiler for C supports arguments -Wno-unused-but-set-variable: YES
  Compiler for C supports arguments -Wno-unused-function: YES
  Compiler for C supports arguments -Wno-conversion: YES
  Compiler for C supports arguments -Wno-misleading-indentation: YES
  Library m found: YES
  Fortran compiler for the host machine: gfortran (gcc 10.3.0 "GNU Fortran (Built by Jeroen for the R-project) 10.3.0")
  Fortran linker for the host machine: gfortran ld.bfd 2.36.1
  ..\..\meson.build:94: WARNING: Consider using the built-in option for language standard version instead of using "-std=legacy".
  Compiler for Fortran supports arguments -Wno-conversion: YES
  Checking if "-Wl,--version-script" links: YES
  Program tools/generate_f2pymod.py found: YES (C:\Users\matth\.local\share\mamba\envs\scipy-dev-numpy126\python.exe C:\Users\matth\Desktop\scipy\tools/generate_f2pymod.py)
  Program scipy/_build_utils/tempita.py found: YES (C:\Users\matth\.local\share\mamba\envs\scipy-dev-numpy126\python.exe C:\Users\matth\Desktop\scipy\scipy/_build_utils/tempita.py)
  Program pythran found: YES 0.18.0 0.18.0 (C:\Users\matth\.local\share\mamba\envs\scipy-dev-numpy126\Scripts\pythran.EXE)
  Found pkg-config: YES (C:\Users\matth\.local\share\mamba\envs\scipy-dev-numpy126\Library\bin\pkg-config.EXE) 0.29.2
  Did not find CMake 'cmake'
  Found CMake: NO
  Run-time dependency xsimd found: NO (tried pkgconfig and cmake)

  Executing subproject xsf

  xsf| Project name: xsf
  xsf| Project version: 0.1.0
  xsf| C++ compiler for the host machine: c++ (gcc 10.3.0 "c++ (Built by Jeroen for the R-project) 10.3.0")
  xsf| C++ linker for the host machine: c++ ld.bfd 2.36.1
  xsf| Build targets in project: 0
  xsf| Subproject xsf finished.


  Executing subproject boost_math

  boost_math| Project name: boost-math
  boost_math| Project version: 1.88.0
  boost_math| Build targets in project: 0
  boost_math| Subproject boost_math finished.


  Executing subproject qhull_r

  qhull_r| Project name: qhull_r
  qhull_r| Project version: 8.0.2
  qhull_r| C compiler for the host machine: cc (gcc 10.3.0 "cc (Built by Jeroen for the R-project) 10.3.0")
  qhull_r| C linker for the host machine: cc ld.bfd 2.36.1
  qhull_r| Compiler for C supports arguments -Wno-unused-but-set-variable: YES (cached)
  qhull_r| Build targets in project: 1
  qhull_r| Subproject qhull_r finished.

  Run-time dependency threads found: YES
  numpy-config found: YES (C:\Users\matth\miniforge3\Scripts\numpy-config.EXE) 2.3.1
  Run-time dependency numpy found: YES 2.3.1
  Library npymath found: YES
  Run-time dependency pybind11 found: YES 2.13.6
  Checking if "thread_local" compiles: NO
  Checking if "_Thread_local" compiles: YES
  Checking if "__thread" compiles: YES
  Configuring scipy_config.h using configuration
  Program f2py found: YES (C:\Users\matth\.local\share\mamba\envs\scipy-dev-numpy126\Scripts\f2py.EXE)
  Run-time dependency scipy-openblas found: NO (tried pkgconfig)
  Run-time dependency openblas found: YES 0.3.30
  Dependency openblas found: YES 0.3.30 (cached)
  Compiler for C supports arguments -Wno-maybe-uninitialized: YES
  Compiler for C supports arguments -Wno-discarded-qualifiers: YES
  Compiler for C supports arguments -Wno-empty-body: YES
  Compiler for C supports arguments -Wno-implicit-function-declaration: YES
  Compiler for C supports arguments -Wno-parentheses: YES
  Compiler for C supports arguments -Wno-switch: YES
  Compiler for C supports arguments -Wno-unused-label: YES
  Compiler for C supports arguments -Wno-unused-result: YES
  Compiler for C supports arguments -Wno-unused-variable: YES
  Compiler for C supports arguments -Wno-unused-but-set-variable: YES (cached)
  Compiler for C++ supports arguments -Wno-bitwise-instead-of-logical: NO
  Compiler for C++ supports arguments -Wno-cpp: YES
  Compiler for C++ supports arguments -Wno-class-memaccess: YES
  Compiler for C++ supports arguments -Wno-deprecated-declarations: YES
  Compiler for C++ supports arguments -Wno-deprecated-builtins: NO
  Compiler for C++ supports arguments -Wno-format-truncation: YES
  Compiler for C++ supports arguments -Wno-non-virtual-dtor: YES
  Compiler for C++ supports arguments -Wno-sign-compare: YES
  Compiler for C++ supports arguments -Wno-switch: YES
  Compiler for C++ supports arguments -Wno-terminate: YES
  Compiler for C++ supports arguments -Wno-unused-but-set-variable: YES
  Compiler for C++ supports arguments -Wno-unused-function: YES
  Compiler for C++ supports arguments -Wno-unused-local-typedefs: YES
  Compiler for C++ supports arguments -Wno-unused-variable: YES
  Compiler for C++ supports arguments -Wno-int-in-bool-context: YES
  Compiler for Fortran supports arguments -Wno-argument-mismatch: YES
  Compiler for Fortran supports arguments -Wno-conversion: YES (cached)
  Compiler for Fortran supports arguments -Wno-intrinsic-shadow: YES
  Compiler for Fortran supports arguments -Wno-maybe-uninitialized: YES
  Compiler for Fortran supports arguments -Wno-surprising: YES
  Compiler for Fortran supports arguments -Wno-uninitialized: YES
  Compiler for Fortran supports arguments -Wno-unused-dummy-argument: YES
  Compiler for Fortran supports arguments -Wno-unused-label: YES
  Compiler for Fortran supports arguments -Wno-unused-variable: YES
  Compiler for Fortran supports arguments -Wno-tabs: YES
  Compiler for Fortran supports arguments -Wno-argument-mismatch: YES (cached)
  Compiler for Fortran supports arguments -Wno-conversion: YES (cached)
  Compiler for Fortran supports arguments -Wno-maybe-uninitialized: YES (cached)
  Compiler for Fortran supports arguments -Wno-unused-dummy-argument: YES (cached)
  Compiler for Fortran supports arguments -Wno-unused-label: YES (cached)
  Compiler for Fortran supports arguments -Wno-unused-variable: YES (cached)
  Compiler for Fortran supports arguments -Wno-tabs: YES (cached)
  Checking if "Check atomic builtins without -latomic" links: YES
  Configuring __config__.py using configuration
  Checking for function "open_memstream" : NO
  Configuring messagestream_config.h using configuration
  Program _generate_pyx.py found: YES (C:\Users\matth\.local\share\mamba\envs\scipy-dev-numpy126\python.exe C:\Users\matth\Desktop\scipy\scipy\special\_generate_pyx.py)
  Program _generate_pyx.py found: YES (C:\Users\matth\.local\share\mamba\envs\scipy-dev-numpy126\python.exe C:\Users\matth\Desktop\scipy\scipy\linalg\_generate_pyx.py)
  Program ../_generate_sparsetools.py found: YES (C:\Users\matth\.local\share\mamba\envs\scipy-dev-numpy126\python.exe C:\Users\matth\Desktop\scipy\scipy\sparse\sparsetools\../_generate_sparsetools.py)
  Checking for size of "void*" : 8
  Compiler for Fortran supports arguments -w: YES
  Checking for size of "void*" : 8

  Executing subproject highs

  highs| Project name: highs
  highs| Project version: 1.8.0
  highs| C compiler for the host machine: cc (gcc 10.3.0 "cc (Built by Jeroen for the R-project) 10.3.0")
  highs| C linker for the host machine: cc ld.bfd 2.36.1
  highs| C++ compiler for the host machine: c++ (gcc 10.3.0 "c++ (Built by Jeroen for the R-project) 10.3.0")
  highs| C++ linker for the host machine: c++ ld.bfd 2.36.1
  highs| Compiler for C++ supports arguments -Wno-invalid-offsetof: YES
  highs| Compiler for C++ supports arguments -Wno-maybe-uninitialized: YES
  highs| Compiler for C++ supports arguments -Wno-reorder: YES
  highs| Compiler for C++ supports arguments -Wno-reorder-ctor: NO
  highs| Compiler for C++ supports arguments -Wno-sometimes-uninitialized: NO
  highs| Compiler for C++ supports arguments -Wno-unused-but-set-variable: YES (cached)
  highs| Compiler for C++ supports arguments -Wno-unused-variable: YES (cached)
  highs| Compiler for C++ supports arguments -Wno-use-after-free: NO
  highs| Compiler for C++ supports arguments -Wno-comment: YES
  highs| Compiler for C supports arguments -Wno-comment: YES
  highs| Compiler for C supports arguments -Wno-invalid-offsetof: NO
  highs| Compiler for C supports arguments -Wno-maybe-uninitialized: YES (cached)
  highs| Compiler for C supports arguments -Wno-sometimes-uninitialized: NO
  highs| Compiler for C supports arguments -Wno-unused-label: YES (cached)
  highs| Compiler for C supports arguments -Wno-use-after-free: NO
  highs| Compiler for C supports arguments -Wno-unused-but-set-variable: YES (cached)
  highs| Compiler for C supports arguments -Wno-unused-variable: YES (cached)
  highs| Compiler for C supports arguments -Wno-use-after-free: NO (cached)
  highs| Dependency threads found: YES unknown (cached)
  highs| Checking if "Check atomic builtins without -latomic" links: YES (cached)
  highs| Dependency zlib skipped: feature use_zlib disabled
  highs| Checking if "mm_pause check" compiles: YES
  highs| Checking if "builtin_clz check" compiles: YES
  highs| Configuring HConfig.h.meson.interim using configuration
  highs| Found git repository at C:\Users\matth\Desktop\scipy
  highs| Build targets in project: 145
  highs| Subproject highs finished.

  Build targets in project: 195

  scipy 1.16.1.dev0+git20250623.f66c2ff

    Subprojects
      boost_math  : YES
      highs       : YES
      qhull_r     : YES
      xsf         : YES

    User defined options
      Native files: C:\Users\matth\Desktop\scipy\build\cp312\meson-python-native-file.ini
      b_ndebug    : if-release
      b_vscrt     : md
      buildtype   : release

  Found ninja.EXE-1.13.1 at C:\Users\matth\.local\share\mamba\envs\scipy-dev-numpy126\Library\bin\ninja.EXE

  Generating targets:   0%|          | 0/195 eta ?
  Generating targets:  25%|##5       | 49/195 eta 00:00
  Generating targets:  50%|####9     | 97/195 eta 00:00
  Generating targets:  66%|######6   | 129/195 eta 00:00
  Generating targets:  81%|########  | 157/195 eta 00:00
  Generating targets:  94%|#########4| 184/195 eta 00:00


  Writing build.ninja:   0%|          | 0/1615 eta ?
  Writing build.ninja:  59%|#####9    | 956/1615 eta 00:00

  + meson compile
  [1/1469] Copying file scipy/optimize/cython_optimize/__init__.py
  [2/1469] Scanning target "lib__dpropack" for modules
  [3/1469] Scanning target "lsoda_lib" for modules
  [4/1469] Scanning target "mach_lib" for modules
  [5/1469] Scanning target "lib__spropack" for modules
  [6/1469] Scanning target "lib__zpropack" for modules
  [7/1469] Scanning target "lib__cpropack" for modules
  [8/1469] Scanning target "dop_lib" for modules
  [9/1469] Scanning target "vode_lib" for modules
  [10/1469] Generating scipy/sparse/linalg/_eigen/arpack/arpack_module with a custom command
  [11/1469] Generating from 'vode.pyf'
  [12/1469] Scanning target "arpack_lib" for modules
  [13/1469] Generating from 'test_fortran.pyf'
  [14/1469] Generating from 'zpropack.pyf'
  [15/1469] Copying file scipy/__init__.py
  [16/1469] Copying file scipy/_lib/__init__.py
  [17/1469] Generating from 'cpropack.pyf'
  [18/1469] Generating from 'spropack.pyf'
  [19/1469] Generating from 'dpropack.pyf'
  [20/1469] Copying file scipy/_lib/ccallback.pxd
  [21/1469] Copying file scipy/_lib/_ccallback_c.pxd
  [22/1469] Copying file scipy/_lib/messagestream.pxd
  [23/1469] Copying file scipy/special/__init__.py
  [24/1469] Copying file scipy/special/_agm.pxd
  [25/1469] Generating from 'lsoda.pyf'
  [26/1469] Copying file scipy/special/_boxcox.pxd
  [27/1469] Copying file scipy/special/_complexstuff.pxd
  [28/1469] Scanning target "fitpack_lib" for modules
  [29/1469] Copying file scipy/special/_ellip_harm_2.pxd
  [30/1469] Copying file scipy/special/_convex_analysis.pxd
  [31/1469] Copying file scipy/special/_ellipk.pxd
  [32/1469] Copying file scipy/special/_ellip_harm.pxd
  [33/1469] Copying file scipy/special/_factorial.pxd
  [34/1469] Copying file scipy/special/_cdflib_wrappers.pxd
  [35/1469] Copying file scipy/special/_hyp0f1.pxd
  [36/1469] Copying file scipy/special/_hypergeometric.pxd
  [37/1469] Scanning target "odrpack" for modules
  [38/1469] Generating from 'dop.pyf'
  [39/1469] Generating from 'tests/test_odeint_banded.pyf'
  [40/1469] Generating scipy/special/cython_special with a custom command
  [41/1469] Copying file scipy/special/_legacy.pxd
  [42/1469] Copying file scipy/special/_ufuncs_extra_code.pxi
  [43/1469] Copying file scipy/special/_ndtri_exp.pxd
  [44/1469] Copying file scipy/special/_sici.pxd
  [45/1469] Copying file scipy/special/_spence.pxd
  [46/1469] Generating from 'src/dfitpack.pyf'
  [47/1469] Copying file scipy/special/cython_special.pxd
  [48/1469] Copying file scipy/special/sf_error.pxd
  [49/1469] Copying file scipy/special/_ufuncs_extra_code_common.pxi
  [50/1469] Copying file scipy/special/orthogonal_eval.pxd
  [51/1469] Copying file scipy/linalg/__init__.py
  [52/1469] Generating scipy/linalg/_decomp_update with a custom command
  [53/1469] Generating scipy/sparse/_csparsetools_pyx with a custom command
  [54/1469] Generating scipy/linalg/cython_linalg with a custom command
  [55/1469] Copying file scipy/special/__init__.pxd
  [56/1469] Copying file scipy/linalg/_cythonized_array_utils.pxd
  [57/1469] Copying file scipy/linalg/__init__.pxd
  [58/1469] Generating dynamic dependency information for target "lib__spropack"
  [59/1469] Copying file scipy/sparse/csgraph/parameters.pxi
  [60/1469] Generating dynamic dependency information for target "lib__cpropack"
  [61/1469] Generating dynamic dependency information for target "lib__dpropack"
  [62/1469] Generating dynamic dependency information for target "lib__zpropack"
  [63/1469] Copying file scipy/stats/__init__.py
  [64/1469] Copying file scipy/stats/_stats.pxd
  [65/1469] Generating dynamic dependency information for target "arpack_lib"
  [66/1469] Generating scipy/_cyutility.c with a custom command
  [67/1469] Copying file scipy/stats/_unuran/unuran.pxd
  [68/1469] Copying file scipy/stats/_unuran/__init__.py
  [69/1469] Copying file scipy/stats/_biasedurn.pxd
  [70/1469] Generating scipy/optimize/cython_optimize/_zeros_pyx with a custom command
  [71/1469] Copying file scipy/optimize/__init__.py
  [72/1469] Copying file scipy/optimize/cython_optimize/_zeros.pxd
  [73/1469] Scanning target "_cpropack.cp312-win_amd64" for modules
  FAILED: [code=1] scipy/sparse/linalg/_propack/_cpropack.cp312-win_amd64.pyd.p/depscan.json
  "C:\Users\matth\.local\share\mamba\envs\scipy-dev-numpy126\Scripts\meson" "--internal" "depscan"  scipy/sparse/linalg/_propack/_cpropack.cp312-win_amd64.pyd.p/depscan.json scipy/sparse/linalg/_propack/_cpropack.cp312-win_amd64.pyd.p/_cpropack.cp312-win_amd64.dat
  Traceback (most recent call last):
    File "C:\Users\matth\.local\share\mamba\envs\scipy-dev-numpy126\Scripts\meson-script.py", line 10, in <module>
      sys.exit(main())
               ^^^^^^
    File "C:\Users\matth\.local\share\mamba\envs\scipy-dev-numpy126\Lib\site-packages\mesonbuild\mesonmain.py", line 313, in main
      return run(sys.argv[1:], launcher)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^
    File "C:\Users\matth\.local\share\mamba\envs\scipy-dev-numpy126\Lib\site-packages\mesonbuild\mesonmain.py", line 300, in run
      return run_script_command(args[1], args[2:])
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    File "C:\Users\matth\.local\share\mamba\envs\scipy-dev-numpy126\Lib\site-packages\mesonbuild\mesonmain.py", line 222, in run_script_command
      return module.run(script_args)
             ^^^^^^^^^^^^^^^^^^^^^^^
    File "C:\Users\matth\.local\share\mamba\envs\scipy-dev-numpy126\Lib\site-packages\mesonbuild\scripts\depscan.py", line 208, in run
      return scanner.scan()
             ^^^^^^^^^^^^^^
    File "C:\Users\matth\.local\share\mamba\envs\scipy-dev-numpy126\Lib\site-packages\mesonbuild\scripts\depscan.py", line 163, in scan
      self.scan_file(s, lang)
    File "C:\Users\matth\.local\share\mamba\envs\scipy-dev-numpy126\Lib\site-packages\mesonbuild\scripts\depscan.py", line 83, in scan_file
      self.scan_fortran_file(fname)
    File "C:\Users\matth\.local\share\mamba\envs\scipy-dev-numpy126\Lib\site-packages\mesonbuild\scripts\depscan.py", line 90, in scan_fortran_file
      for line in fpath.read_text(encoding='utf-8', errors='ignore').split('\n'):
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    File "C:\Users\matth\.local\share\mamba\envs\scipy-dev-numpy126\Lib\pathlib.py", line 1027, in read_text
      with self.open(mode='r', encoding=encoding, errors=errors) as f:
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    File "C:\Users\matth\.local\share\mamba\envs\scipy-dev-numpy126\Lib\pathlib.py", line 1013, in open
      return io.open(self, mode, buffering, encoding, errors, newline)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  FileNotFoundError: [Errno 2] No such file or directory: 'scipy\\sparse\\linalg\\_propack\\_cpropack.cp312-win_amd64.pyd.p\\_cpropack-f2pywrappers.f'
  [74/1469] Copying file scipy/optimize/__init__.pxd
  [75/1469] Scanning target "_dpropack.cp312-win_amd64" for modules
  FAILED: [code=1] scipy/sparse/linalg/_propack/_dpropack.cp312-win_amd64.pyd.p/depscan.json
  "C:\Users\matth\.local\share\mamba\envs\scipy-dev-numpy126\Scripts\meson" "--internal" "depscan"  scipy/sparse/linalg/_propack/_dpropack.cp312-win_amd64.pyd.p/depscan.json scipy/sparse/linalg/_propack/_dpropack.cp312-win_amd64.pyd.p/_dpropack.cp312-win_amd64.dat
  Traceback (most recent call last):
    File "C:\Users\matth\.local\share\mamba\envs\scipy-dev-numpy126\Scripts\meson-script.py", line 10, in <module>
      sys.exit(main())
               ^^^^^^
    File "C:\Users\matth\.local\share\mamba\envs\scipy-dev-numpy126\Lib\site-packages\mesonbuild\mesonmain.py", line 313, in main
      return run(sys.argv[1:], launcher)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^
    File "C:\Users\matth\.local\share\mamba\envs\scipy-dev-numpy126\Lib\site-packages\mesonbuild\mesonmain.py", line 300, in run
      return run_script_command(args[1], args[2:])
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    File "C:\Users\matth\.local\share\mamba\envs\scipy-dev-numpy126\Lib\site-packages\mesonbuild\mesonmain.py", line 222, in run_script_command
      return module.run(script_args)
             ^^^^^^^^^^^^^^^^^^^^^^^
    File "C:\Users\matth\.local\share\mamba\envs\scipy-dev-numpy126\Lib\site-packages\mesonbuild\scripts\depscan.py", line 208, in run
      return scanner.scan()
             ^^^^^^^^^^^^^^
    File "C:\Users\matth\.local\share\mamba\envs\scipy-dev-numpy126\Lib\site-packages\mesonbuild\scripts\depscan.py", line 163, in scan
      self.scan_file(s, lang)
    File "C:\Users\matth\.local\share\mamba\envs\scipy-dev-numpy126\Lib\site-packages\mesonbuild\scripts\depscan.py", line 83, in scan_file
      self.scan_fortran_file(fname)
    File "C:\Users\matth\.local\share\mamba\envs\scipy-dev-numpy126\Lib\site-packages\mesonbuild\scripts\depscan.py", line 90, in scan_fortran_file
      for line in fpath.read_text(encoding='utf-8', errors='ignore').split('\n'):
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    File "C:\Users\matth\.local\share\mamba\envs\scipy-dev-numpy126\Lib\pathlib.py", line 1027, in read_text
      with self.open(mode='r', encoding=encoding, errors=errors) as f:
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    File "C:\Users\matth\.local\share\mamba\envs\scipy-dev-numpy126\Lib\pathlib.py", line 1013, in open
      return io.open(self, mode, buffering, encoding, errors, newline)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  FileNotFoundError: [Errno 2] No such file or directory: 'scipy\\sparse\\linalg\\_propack\\_dpropack.cp312-win_amd64.pyd.p\\_dpropack-f2pywrappers.f'
  [76/1469] Copying file scipy/optimize/cython_optimize/c_zeros.pxd
  [77/1469] Scanning target "_zpropack.cp312-win_amd64" for modules
  FAILED: [code=1] scipy/sparse/linalg/_propack/_zpropack.cp312-win_amd64.pyd.p/depscan.json
  "C:\Users\matth\.local\share\mamba\envs\scipy-dev-numpy126\Scripts\meson" "--internal" "depscan"  scipy/sparse/linalg/_propack/_zpropack.cp312-win_amd64.pyd.p/depscan.json scipy/sparse/linalg/_propack/_zpropack.cp312-win_amd64.pyd.p/_zpropack.cp312-win_amd64.dat
  Traceback (most recent call last):
    File "C:\Users\matth\.local\share\mamba\envs\scipy-dev-numpy126\Scripts\meson-script.py", line 10, in <module>
      sys.exit(main())
               ^^^^^^
    File "C:\Users\matth\.local\share\mamba\envs\scipy-dev-numpy126\Lib\site-packages\mesonbuild\mesonmain.py", line 313, in main
      return run(sys.argv[1:], launcher)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^
    File "C:\Users\matth\.local\share\mamba\envs\scipy-dev-numpy126\Lib\site-packages\mesonbuild\mesonmain.py", line 300, in run
      return run_script_command(args[1], args[2:])
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    File "C:\Users\matth\.local\share\mamba\envs\scipy-dev-numpy126\Lib\site-packages\mesonbuild\mesonmain.py", line 222, in run_script_command
      return module.run(script_args)
             ^^^^^^^^^^^^^^^^^^^^^^^
    File "C:\Users\matth\.local\share\mamba\envs\scipy-dev-numpy126\Lib\site-packages\mesonbuild\scripts\depscan.py", line 208, in run
      return scanner.scan()
             ^^^^^^^^^^^^^^
    File "C:\Users\matth\.local\share\mamba\envs\scipy-dev-numpy126\Lib\site-packages\mesonbuild\scripts\depscan.py", line 163, in scan
      self.scan_file(s, lang)
    File "C:\Users\matth\.local\share\mamba\envs\scipy-dev-numpy126\Lib\site-packages\mesonbuild\scripts\depscan.py", line 83, in scan_file
      self.scan_fortran_file(fname)
    File "C:\Users\matth\.local\share\mamba\envs\scipy-dev-numpy126\Lib\site-packages\mesonbuild\scripts\depscan.py", line 90, in scan_fortran_file
      for line in fpath.read_text(encoding='utf-8', errors='ignore').split('\n'):
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    File "C:\Users\matth\.local\share\mamba\envs\scipy-dev-numpy126\Lib\pathlib.py", line 1027, in read_text
      with self.open(mode='r', encoding=encoding, errors=errors) as f:
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    File "C:\Users\matth\.local\share\mamba\envs\scipy-dev-numpy126\Lib\pathlib.py", line 1013, in open
      return io.open(self, mode, buffering, encoding, errors, newline)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  FileNotFoundError: [Errno 2] No such file or directory: 'scipy\\sparse\\linalg\\_propack\\_zpropack.cp312-win_amd64.pyd.p\\_zpropack-f2pywrappers.f'
  [78/1469] Scanning target "_arpack.cp312-win_amd64" for modules
  FAILED: [code=1] scipy/sparse/linalg/_eigen/arpack/_arpack.cp312-win_amd64.pyd.p/depscan.json
  "C:\Users\matth\.local\share\mamba\envs\scipy-dev-numpy126\Scripts\meson" "--internal" "depscan"  scipy/sparse/linalg/_eigen/arpack/_arpack.cp312-win_amd64.pyd.p/depscan.json scipy/sparse/linalg/_eigen/arpack/_arpack.cp312-win_amd64.pyd.p/_arpack.cp312-win_amd64.dat
  Traceback (most recent call last):
    File "C:\Users\matth\.local\share\mamba\envs\scipy-dev-numpy126\Scripts\meson-script.py", line 10, in <module>
      sys.exit(main())
               ^^^^^^
    File "C:\Users\matth\.local\share\mamba\envs\scipy-dev-numpy126\Lib\site-packages\mesonbuild\mesonmain.py", line 313, in main
      return run(sys.argv[1:], launcher)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^
    File "C:\Users\matth\.local\share\mamba\envs\scipy-dev-numpy126\Lib\site-packages\mesonbuild\mesonmain.py", line 300, in run
      return run_script_command(args[1], args[2:])
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    File "C:\Users\matth\.local\share\mamba\envs\scipy-dev-numpy126\Lib\site-packages\mesonbuild\mesonmain.py", line 222, in run_script_command
      return module.run(script_args)
             ^^^^^^^^^^^^^^^^^^^^^^^
    File "C:\Users\matth\.local\share\mamba\envs\scipy-dev-numpy126\Lib\site-packages\mesonbuild\scripts\depscan.py", line 208, in run
      return scanner.scan()
             ^^^^^^^^^^^^^^
    File "C:\Users\matth\.local\share\mamba\envs\scipy-dev-numpy126\Lib\site-packages\mesonbuild\scripts\depscan.py", line 163, in scan
      self.scan_file(s, lang)
    File "C:\Users\matth\.local\share\mamba\envs\scipy-dev-numpy126\Lib\site-packages\mesonbuild\scripts\depscan.py", line 83, in scan_file
      self.scan_fortran_file(fname)
    File "C:\Users\matth\.local\share\mamba\envs\scipy-dev-numpy126\Lib\site-packages\mesonbuild\scripts\depscan.py", line 90, in scan_fortran_file
      for line in fpath.read_text(encoding='utf-8', errors='ignore').split('\n'):
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    File "C:\Users\matth\.local\share\mamba\envs\scipy-dev-numpy126\Lib\pathlib.py", line 1027, in read_text
      with self.open(mode='r', encoding=encoding, errors=errors) as f:
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    File "C:\Users\matth\.local\share\mamba\envs\scipy-dev-numpy126\Lib\pathlib.py", line 1013, in open
      return io.open(self, mode, buffering, encoding, errors, newline)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  FileNotFoundError: [Errno 2] No such file or directory: 'scipy\\sparse\\linalg\\_eigen\\arpack\\_arpack-f2pywrappers.f'
  [79/1469] Scanning target "_spropack.cp312-win_amd64" for modules
  FAILED: [code=1] scipy/sparse/linalg/_propack/_spropack.cp312-win_amd64.pyd.p/depscan.json
  "C:\Users\matth\.local\share\mamba\envs\scipy-dev-numpy126\Scripts\meson" "--internal" "depscan"  scipy/sparse/linalg/_propack/_spropack.cp312-win_amd64.pyd.p/depscan.json scipy/sparse/linalg/_propack/_spropack.cp312-win_amd64.pyd.p/_spropack.cp312-win_amd64.dat
  Traceback (most recent call last):
    File "C:\Users\matth\.local\share\mamba\envs\scipy-dev-numpy126\Scripts\meson-script.py", line 10, in <module>
      sys.exit(main())
               ^^^^^^
    File "C:\Users\matth\.local\share\mamba\envs\scipy-dev-numpy126\Lib\site-packages\mesonbuild\mesonmain.py", line 313, in main
      return run(sys.argv[1:], launcher)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^
    File "C:\Users\matth\.local\share\mamba\envs\scipy-dev-numpy126\Lib\site-packages\mesonbuild\mesonmain.py", line 300, in run
      return run_script_command(args[1], args[2:])
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    File "C:\Users\matth\.local\share\mamba\envs\scipy-dev-numpy126\Lib\site-packages\mesonbuild\mesonmain.py", line 222, in run_script_command
      return module.run(script_args)
             ^^^^^^^^^^^^^^^^^^^^^^^
    File "C:\Users\matth\.local\share\mamba\envs\scipy-dev-numpy126\Lib\site-packages\mesonbuild\scripts\depscan.py", line 208, in run
      return scanner.scan()
             ^^^^^^^^^^^^^^
    File "C:\Users\matth\.local\share\mamba\envs\scipy-dev-numpy126\Lib\site-packages\mesonbuild\scripts\depscan.py", line 163, in scan
      self.scan_file(s, lang)
    File "C:\Users\matth\.local\share\mamba\envs\scipy-dev-numpy126\Lib\site-packages\mesonbuild\scripts\depscan.py", line 83, in scan_file
      self.scan_fortran_file(fname)
    File "C:\Users\matth\.local\share\mamba\envs\scipy-dev-numpy126\Lib\site-packages\mesonbuild\scripts\depscan.py", line 90, in scan_fortran_file
      for line in fpath.read_text(encoding='utf-8', errors='ignore').split('\n'):
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    File "C:\Users\matth\.local\share\mamba\envs\scipy-dev-numpy126\Lib\pathlib.py", line 1027, in read_text
      with self.open(mode='r', encoding=encoding, errors=errors) as f:
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    File "C:\Users\matth\.local\share\mamba\envs\scipy-dev-numpy126\Lib\pathlib.py", line 1013, in open
      return io.open(self, mode, buffering, encoding, errors, newline)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  FileNotFoundError: [Errno 2] No such file or directory: 'scipy\\sparse\\linalg\\_propack\\_spropack.cp312-win_amd64.pyd.p\\_spropack-f2pywrappers.f'
  [80/1469] Copying file scipy/optimize/cython_optimize.pxd
  [81/1469] Copying file scipy/spatial/_qhull.pxd
  [82/1469] Scanning target "_test_fortran.cp312-win_amd64" for modules
  FAILED: [code=1] scipy/io/_test_fortran.cp312-win_amd64.pyd.p/depscan.json
  "C:\Users\matth\.local\share\mamba\envs\scipy-dev-numpy126\Scripts\meson" "--internal" "depscan"  scipy/io/_test_fortran.cp312-win_amd64.pyd.p/depscan.json scipy/io/_test_fortran.cp312-win_amd64.pyd.p/_test_fortran.cp312-win_amd64.dat
  Traceback (most recent call last):
    File "C:\Users\matth\.local\share\mamba\envs\scipy-dev-numpy126\Scripts\meson-script.py", line 10, in <module>
      sys.exit(main())
               ^^^^^^
    File "C:\Users\matth\.local\share\mamba\envs\scipy-dev-numpy126\Lib\site-packages\mesonbuild\mesonmain.py", line 313, in main
      return run(sys.argv[1:], launcher)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^
    File "C:\Users\matth\.local\share\mamba\envs\scipy-dev-numpy126\Lib\site-packages\mesonbuild\mesonmain.py", line 300, in run
      return run_script_command(args[1], args[2:])
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    File "C:\Users\matth\.local\share\mamba\envs\scipy-dev-numpy126\Lib\site-packages\mesonbuild\mesonmain.py", line 222, in run_script_command
      return module.run(script_args)
             ^^^^^^^^^^^^^^^^^^^^^^^
    File "C:\Users\matth\.local\share\mamba\envs\scipy-dev-numpy126\Lib\site-packages\mesonbuild\scripts\depscan.py", line 208, in run
      return scanner.scan()
             ^^^^^^^^^^^^^^
    File "C:\Users\matth\.local\share\mamba\envs\scipy-dev-numpy126\Lib\site-packages\mesonbuild\scripts\depscan.py", line 163, in scan
      self.scan_file(s, lang)
    File "C:\Users\matth\.local\share\mamba\envs\scipy-dev-numpy126\Lib\site-packages\mesonbuild\scripts\depscan.py", line 83, in scan_file
      self.scan_fortran_file(fname)
    File "C:\Users\matth\.local\share\mamba\envs\scipy-dev-numpy126\Lib\site-packages\mesonbuild\scripts\depscan.py", line 90, in scan_fortran_file
      for line in fpath.read_text(encoding='utf-8', errors='ignore').split('\n'):
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    File "C:\Users\matth\.local\share\mamba\envs\scipy-dev-numpy126\Lib\pathlib.py", line 1027, in read_text
      with self.open(mode='r', encoding=encoding, errors=errors) as f:
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    File "C:\Users\matth\.local\share\mamba\envs\scipy-dev-numpy126\Lib\pathlib.py", line 1013, in open
      return io.open(self, mode, buffering, encoding, errors, newline)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  FileNotFoundError: [Errno 2] No such file or directory: 'scipy\\io\\_test_fortran.cp312-win_amd64.pyd.p\\_test_fortran-f2pywrappers.f'
  [83/1469] Generating dynamic dependency information for target "mach_lib"
  [84/1469] Generating dynamic dependency information for target "lsoda_lib"
  [85/1469] Generating subprojects/highs/src/HConfig.h with a custom command
  [86/1469] Copying file scipy/spatial/setlist.pxd
  [87/1469] Generating dynamic dependency information for target "dop_lib"
  [88/1469] Generating dynamic dependency information for target "vode_lib"
  [89/1469] Scanning target "_vode.cp312-win_amd64" for modules
  FAILED: [code=1] scipy/integrate/_vode.cp312-win_amd64.pyd.p/depscan.json
  "C:\Users\matth\.local\share\mamba\envs\scipy-dev-numpy126\Scripts\meson" "--internal" "depscan"  scipy/integrate/_vode.cp312-win_amd64.pyd.p/depscan.json scipy/integrate/_vode.cp312-win_amd64.pyd.p/_vode.cp312-win_amd64.dat
  Traceback (most recent call last):
    File "C:\Users\matth\.local\share\mamba\envs\scipy-dev-numpy126\Scripts\meson-script.py", line 10, in <module>
      sys.exit(main())
               ^^^^^^
    File "C:\Users\matth\.local\share\mamba\envs\scipy-dev-numpy126\Lib\site-packages\mesonbuild\mesonmain.py", line 313, in main
      return run(sys.argv[1:], launcher)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^
    File "C:\Users\matth\.local\share\mamba\envs\scipy-dev-numpy126\Lib\site-packages\mesonbuild\mesonmain.py", line 300, in run
      return run_script_command(args[1], args[2:])
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    File "C:\Users\matth\.local\share\mamba\envs\scipy-dev-numpy126\Lib\site-packages\mesonbuild\mesonmain.py", line 222, in run_script_command
      return module.run(script_args)
             ^^^^^^^^^^^^^^^^^^^^^^^
    File "C:\Users\matth\.local\share\mamba\envs\scipy-dev-numpy126\Lib\site-packages\mesonbuild\scripts\depscan.py", line 208, in run
      return scanner.scan()
             ^^^^^^^^^^^^^^
    File "C:\Users\matth\.local\share\mamba\envs\scipy-dev-numpy126\Lib\site-packages\mesonbuild\scripts\depscan.py", line 163, in scan
      self.scan_file(s, lang)
    File "C:\Users\matth\.local\share\mamba\envs\scipy-dev-numpy126\Lib\site-packages\mesonbuild\scripts\depscan.py", line 83, in scan_file
      self.scan_fortran_file(fname)
    File "C:\Users\matth\.local\share\mamba\envs\scipy-dev-numpy126\Lib\site-packages\mesonbuild\scripts\depscan.py", line 90, in scan_fortran_file
      for line in fpath.read_text(encoding='utf-8', errors='ignore').split('\n'):
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    File "C:\Users\matth\.local\share\mamba\envs\scipy-dev-numpy126\Lib\pathlib.py", line 1027, in read_text
      with self.open(mode='r', encoding=encoding, errors=errors) as f:
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    File "C:\Users\matth\.local\share\mamba\envs\scipy-dev-numpy126\Lib\pathlib.py", line 1013, in open
      return io.open(self, mode, buffering, encoding, errors, newline)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  FileNotFoundError: [Errno 2] No such file or directory: 'scipy\\integrate\\_vode.cp312-win_amd64.pyd.p\\_vode-f2pywrappers.f'
  [90/1469] Compiling C object scipy/_cyutility.cp312-win_amd64.pyd.p/meson-generated_..__cyutility.c.obj
  ninja: build stopped: subcommand failed.
  INFO: autodetecting backend as ninja
  INFO: calculating backend command to run: C:\Users\matth\.local\share\mamba\envs\scipy-dev-numpy126\Library\bin\ninja.EXE
  error: subprocess-exited-with-error

  × Preparing editable metadata (pyproject.toml) did not run successfully.
  │ exit code: 1
  ╰─> See above for output.

  note: This error originates from a subprocess, and is likely not a problem with pip.
  full command: 'C:\Users\matth\.local\share\mamba\envs\scipy-dev-numpy126\python.exe' 'C:\Users\matth\.local\share\mamba\envs\scipy-dev-numpy126\Lib\site-packages\pip\_vendor\pyproject_hooks\_in_process\_in_process.py' prepare_metadata_for_build_editable 'C:\Users\matth\AppData\Local\Temp\tmpaj8v91az'
  cwd: C:\Users\matth\Desktop\scipy
  Preparing editable metadata (pyproject.toml) ... error
error: metadata-generation-failed

× Encountered error while generating package metadata.
╰─> See above for output.

note: This is an issue with the package mentioned above, not pip.
hint: See above for details.
```

</details>

Can someone try this using the various sorting algorithms in `argsort`? I'd like to see if we can reproduce the NumPy 2.0+ results reported here using one of those (`'stable'`)?

---

BTW, thanks @cooperjaXC for the report and debugging!

Note that the documentation does state "There is currently no special handling of ties in `x`; they are broken arbitrarily by the implementation". So there is currently no guarantee that ties will be broken the same way in all cases. The paper recommends "If there are ties among the Xi’s, then choose an increasing rearrangement as above by breaking ties uniformly at random.". If you do that by adding noise to the Xi, the discrepancy disappears. The paper also suggests:

> If the Xi’s have ties, then ξn(X,Y) is a randomized estimate of ξ(X,Y), because of the randomness coming from the breaking of ties. This can be ignored if n is large, because ξn is guaranteed to be close to ξ by Theorem1.1. Alternatively, one can consider taking the average of ξn over all possible increasing rearrangements of the Xi’s.

All possible rearrangements is tricky, but you can get a randomized approximation.

<img width="913" height="418" alt="Image" src="https://github.com/user-attachments/assets/9e64677f-e0f9-4a2b-b001-19ec6158d124" />

--- a/scipy/stats/_correlation.py
+++ b/scipy/stats/_correlation.py
@@ -139,7 +139,7 @@ def chatterjeexi(x, y, *, axis=0, y_continuous=False, method='asymptotic'):
     Notes
     -----
     There is currently no special handling of ties in `x`; they are broken arbitrarily
-    by the implementation. [1]_ recommends: "if there are ties among the Xi’s, then
+    by the implementation. [1]_ recommends: "if there are ties among the Xi's, then
     choose an increasing rearrangement as above by breaking ties uniformly at random."
     This is easily accomplished by adding a small amount of random noise to `x`; see
     examples.

--- a/scipy/stats/_correlation.py
+++ b/scipy/stats/_correlation.py
@@ -139,7 +139,10 @@ def chatterjeexi(x, y, *, axis=0, y_continuous=False, method='asymptotic'):
     Notes
     -----
     There is currently no special handling of ties in `x`; they are broken arbitrarily
-    by the implementation.
+    by the implementation. [1]_ recommends: "if there are ties among the Xi’s, then
+    choose an increasing rearrangement as above by breaking ties uniformly at random."
+    This is easily accomplished by adding a small amount of random noise to `x`; see
+    examples.
 
     [1]_ notes that the statistic is not symmetric in `x` and `y` *by design*:
     "...we may want to understand if :math:`Y` is a function :math:`X`, and not just
@@ -184,6 +187,29 @@ def chatterjeexi(x, y, *, axis=0, y_continuous=False, method='asymptotic'):
 
     >>> stats.chatterjeexi(x, y + noise, y_continuous=True, axis=-1).statistic
     array([0.79507951, 0.41824182, 0.16651665])
+
+    Consider a case in which there are ties in `x`.
+
+    >>> x = rng.integers(10, size=1000)
+    >>> y = rng.integers(10, size=1000)
+
+    [1]_ recommends breaking the ties uniformly at random.
+
+    >>> d = rng.uniform(1e-5, size=x.size)
+    >>> res = stats.chatterjeexi(x + d, y)
+    >>> res.statistic
+    -0.029919991638798438
+
+    Since this gives a randomized estimate of the statistic, [1]_ also suggests
+    considering the average over all possibilities of breaking ties. This is
+    computationally infeasible when there are many ties, but a randomized estimate of
+    *this* quantity can be obtained by considering many random possibilities of breaking
+    ties.
+
+    >>> d = rng.uniform(1e-5, size=(9999, x.size))
+    >>> res = stats.chatterjeexi(x + d, y, axis=1)
+    >>> np.mean(res.statistic)
+    0.001186895213756626
 
     """
     # x, y, `axis` input validation taken care of by decorator

